### PR TITLE
Adding e2e test to verify hyperv container is running inside a VM on Windows

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR adds a new e2e test that starts a Windows hyperv isolated container and then validates the container is running in a VM (through a separately scheduled hostProcessContainer)

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes-sigs/windows-testing/issues/364

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig windows

### CODE REVIEW

File: test/e2e/windows/hyperv.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	f := framework.NewDefaultFramework("windows-hyperv-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	ginkgo.It("should start a hyperv isolated container", func(ctx context.Context) {
+
+		// HyperV isolated containers are only supported on containerd 1.7+
+		skipUnlessContainerdOneSevenOrGreater(ctx, f)
+
+		// check if hyperv runtime class is on node and skip otherwise
+		// note: this test assumes there is a mutating webhook that adds the runtime class
```
Comment (jsturtevant) [inline test/e2e/windows/hyperv.go:47]: It also assumes a runtimeclass with the name `runhcs-wcow-hypervisor`. Is there an example of what this file looks like?

Comment (jsturtevant) [reply test/e2e/windows/hyperv.go:47]: Is there any reason for this specific test we couldn't apply the runtimeclass manually to the pod?

File: test/e2e/windows/hyperv.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+		ginkgo.By("schedule a pod to that node")
+		image := imageutils.GetE2EImage(imageutils.BusyBox)
+		hypervPodName := "hyperv-test-pod"
+		hypervPod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: hypervPodName,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
```
Comment (jsturtevant) [inline test/e2e/windows/hyperv.go:70]: one of the major drawbacks of the previous hyper-v implementation was not able to have more than one container per pod. Should we add an second one here or maybe an second test to validate this?

File: test/e2e/windows/hyperv.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+						HostProcess:   &trueVar,
+						RunAsUserName: &User_NTAuthoritySystem,
+					},
+				},
+				HostNetwork: true,
+				Containers: []v1.Container{
+					{
+						Image:   image,
+						Name:    "container",
+						Command: []string{"powershell.exe", "-Command", "$vms = Get-ComputeProcess | Where-Object { ($_.Type -EQ 'VirtualMachine') -and ($_.Owner -EQ 'containerd-shim-runhcs-v1.exe') } ; if ($vms.Length -le 0) { throw 'error' }"},
```
Comment (jsturtevant) [inline test/e2e/windows/hyperv.go:108]: cool didn't know about `Get-ComputeProcess`

File: test/e2e/windows/hyperv.go (hunk 2 above)
Comment (marosset) [reply test/e2e/windows/hyperv.go:70]: great idea!

File: test/e2e/windows/hyperv.go (hunk 1 above)
Comment (marosset) [reply test/e2e/windows/hyperv.go:47]: We can do that. I just need to make sure the webhook is OK with that (or update the webhook)

Comment (marosset) [reply test/e2e/windows/hyperv.go]: done

**Pull request merged**